engine/client/client.go

```diff
@@ -225,7 +225,7 @@ func Connect(ctx context.Context, params Params) (_ *Client, _ context.Context,
 
 	sessionTask := loader.Task("starting session")
 
-	sharedKey := c.ServerID // share a session across servers
+	sharedKey := ""
```

Comment:
Hey, nice find! Can't believe I didn't notice this one, I've been staring at this file for hours this week :tada:

Ok, so I think this is fine - but is there any reason we couldn't make this the "client unique identifier" (the hostname?). The `sharedKey` here and the `SharedKeyHint` combine to make the `llb.Local` have an isolated space, but is there a reason we couldn't have that info in the `sharedKey` directly? (I think this is all the `sharedKey`s are used for, but I could be missing something).

Also, unrelated, but we should definitely try and have something better than the client hostname. Buildx does a cool thing by combining the directory path and a unique "node identifier", which it stores in `.config/buildx/.buildNodeID`: https://github.com/docker/buildx/blob/8f8ed68b6111552e3d4acda8ba832a735c4c1300/build/build.go#L392

Reply:
> The sharedKey here and the SharedKeyHint combine to make the llb.Local have an isolated space

I feel like just for simplicity purposes it would be better to only use `SharedKeyHint` on the LLB for the cache reuse. As far as I can remember the fact that they are all separate out by `:` here means we can get the same end effect by just changing `SharedKeyHint` and leaving `SharedKey` alone: https://github.com/sipsma/buildkit/blob/89bbb62a1e295391fd55fccc1f86f479de064de1/source/local/source.go#L178-L178 ?

I could be misremembering the details of how that works though. But if that's right, I just like having only one place to think about and just leaving this one empty.

> Also, unrelated, but we should definitely try and have something better than the client hostname. Buildx does a cool thing by combining the directory path and a unique "node identifier", which it stores in .config/buildx/.buildNodeID

Huh yeah, I have some built-in aversion to creating files in our user's homedirs, but it's really not that big of a deal and could be useful here I agree. Not sure how beneficial this ends up being in practice since it would only be useful for laptop/desktop use cases, as opposed to "ephemeral CI runner" cases. And I think hostname is a pretty reasonable heuristic for identity among the laptop/desktop cases, but definitely not flawless. Either way, no opposed, especially if we can re-use code and get it for mostly free 👍